### PR TITLE
tests,cmd/snap-confine: port older snapd-discard-ns tests

### PR DESCRIPTION
**Pull request by zyga**
This patch removes the (unused) spread test from snap-confine directory
and adds the useful parts to the existing spread test for
snap-discard-ns.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: tests/main/snap-discard-ns/task.yaml [hunk 1]
```diff
@@ -7,17 +7,30 @@ details: |
...
+    echo "We can try to discard a namespace before the .mnt file is mounted"
+    touch /run/snapd/ns/test-snapd-tools.mnt
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    echo "We can discard the namespace after a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can discard the namespace after a snap runs"
+    test-snapd-tools.success
+    # The last hex is the same as nsfs but older stat on ubuntu 14.04 doesn't know
+    stat -f -c %T /run/snapd/ns/test-snapd-tools.mnt | MATCH 'proc|nsfs|0x6e736673'
```
Comment (jdstrand) [inline tests/main/snap-discard-ns/task.yaml]: According to https://github.com/snapcore/spread, "MATCH - Run grep -q -e on stdin. Without match, print error including content." If I run this as is, it fails:
```
# 17.04
$ stat -f -c %T /run/snapd/ns/hello-world.mnt
nsfs
$ stat -f -c %T /run/snapd/ns/hello-world.mnt | grep -q -e 'proc|nsfs|0x6e736673' || echo FAIL
FAIL
# 14.04
$ stat -f -c %T /run/snapd/ns/hello-world.mnt
UNKNOWN (0x6e736673)
$ stat -f -c %T /run/snapd/ns/hello-world.mnt | grep -q -e 'proc|nsfs|0x6e736673' || echo FAIL
FAIL
```

You need to backslash the '|'. Eg:
```
$ stat -f -c %T /run/snapd/ns/hello-world.mnt | grep -q -e 'proc\|nsfs\|0x6e736673' && echo PASS
PASS
```

Comment (jdstrand) [inline tests/main/snap-discard-ns/task.yaml]: Also, I understand why 'nsfs' and '0x6e736673', but why 'proc'? If that is intentional, please comment on why.

Comment (zyga) [inline tests/main/snap-discard-ns/task.yaml]: proc is there because on some older kernels there is no nsfs yet.

Comment (zyga) [inline tests/main/snap-discard-ns/task.yaml]: That part of documentation is out of date. Spread's MATCH is actually `grep -q -E "$@"`. If I use `\|` with `-E` the expression no longer works.

Comment (jdstrand) [inline tests/main/snap-discard-ns/task.yaml]: What kernels? What does the output look like? Please document what stat outputs for each of the greps.

Comment (zyga) [inline tests/main/snap-discard-ns/task.yaml]: NOTE: This is not a new test, I'm just re-enabling an existing test.

The affected kernel is, for example, one used on CentOS 7. I tested 3.10.x which doesn't have nsfs yet. The output of stat is similar, except for the filesystem magic number.

**Pull request merged**